optuna/storages/base.py

```diff
@@ -164,8 +165,8 @@ def get_trial(self, trial_id):
         raise NotImplementedError
 
     @abc.abstractmethod
-    def get_all_trials(self, study_id):
-        # type: (int) -> List[structs.FrozenTrial]
+    def get_all_trials(self, study_id, deepcopy=True):
```

Comment:
Why `deepcopy` defaults to `True`?

For backward-compatibility?

Reply:
It's for safety.
Without deep-copying, the trials are shared by caller and callee. That is, if the caller (unintendedly) mutates a part of the returned trials, the consistency of the storage state could break (and vice versa).
When disabling deep-copying, the caller should be aware of the effect and risk, so I think the default value should be `False`.